src/body/mod.rs

```diff
@@ -207,7 +207,13 @@ impl Body {
     /// mut reference to the `Body`, and then use `as_reader()`. It is also possible to
     /// get a non-shared, owned reader via [`Body::into_reader()`].
     ///
-    /// * Reader is not limited. To set a limit use [`Body::with_config()`].
+    /// * Reader is not limited by default. That means a malicious server could
+    ///   exhaust all avaliable memory on your client machine.
+    ///   To set a limit use [`Body::into_with_config()`].
+    /// * Reader will error if `Content-Length` is set, but the connection is closed
+    ///   before all bytes are received.
+    /// * Reader will error if `Transfer-Encoding: chunked` is set, but the connection
+    ///   is closed without a FIN message (graceful connection shutdown) from the server.
```

Comment:
This last point sounds like we potentially could fix. If we have received the chunk ending, we could maybe treat further `io::Error` as unimportant (not pool the connection, but also not surface any error).